lib/gingr/solr_indexer.rb

```diff
@@ -6,20 +6,26 @@
 
 module Gingr
   class SolrIndexer
-    include Config
+    include Logging
 
     attr_accessor :reference_urls
     attr_accessor :solr
 
-    def initialize(solr = nil, reference_urls = nil)
-      solr ||= ENV['SOLR_URL'] || Gingr::Config.default_options[:solr_url]
-      solr = RSolr.connect url: solr, adapter: :net_http_persistent if solr.kind_of? String
-      @solr = solr
-      @reference_urls = reference_urls || {}
+    def initialize(connection = nil, refurls = nil)
+      connection ||= Gingr::Config.getopt(:solr_url)
+      connection = RSolr.connect url: connection, adapter: :net_http_persistent if connection.kind_of? String
+      @solr = connection
+
+      # Strip HTTP Basic Auth
+      @reference_urls = (refurls || {}).transform_values do |url|
+        URI(url).tap { |uri| uri.password = uri.user = nil }.to_s
```

Comment:
This was necessary to ensure that HTTP Basic Auth credentials are always stripped when updating references URLs.